spec/ask_export/runner_spec.rb

```diff
@@ -0,0 +1,96 @@
+        expect { example.run }.to output.to_stdout
+      end
+
+      around do |example|
+        ClimateControl.modify(S3_BUCKET: s3_bucket,
+                              S3_PATH_PREFIX: s3_path_prefix) { example.run }
+      end
+
+      before do
+        allow(AskExport::SurveyResponseFetcher).to receive(:call).and_return(responses)
+        allow(Aws::S3::Resource).to receive(:new).and_return(s3_resource_stub)
+        allow(AskExport::PartnerNotifier).to receive(:call)
+      end
+
+      let(:responses) do
+        [
+          {
+            id: 1,
+            submission_time: "2020-05-01T09:00:00+01:00",
+            region: "Greater London",
+            question: "A question?",
+            question_format: "Read out",
+            name: "Joe Bloggs",
+            email: "joe@example.com",
+            phone: "+44123456789",
+          },
+        ]
+      end
+
+      let(:since_time) { Time.zone.parse("2020-04-30 10:00") }
+      let(:until_time) { Time.zone.parse("2020-05-01 10:00") }
+      let(:s3_bucket) { "s3-bucket" }
+      let(:s3_path_prefix) { "prefix/" }
+      let(:s3_resource_stub) { Aws::S3::Resource.new(stub_responses: true) }
+
+      it "fetches survey responses for the last day with London times" do
+        expect(AskExport::SurveyResponseFetcher)
+          .to receive(:call)
+          .with(since_time, until_time)
+          .and_yield(100)
+          .and_return(responses)
+        described_class.call
+      end
+
+      it "uploads cabinet office and third party CSVs to S3" do
+        csv_builder = instance_double("CsvBuilder",
+                                      cabinet_office: "cabinet-office-data",
+                                      third_party: "third-party-data")
+
+        expect(AskExport::CsvBuilder).to receive(:new).and_return(csv_builder)
+        expect(s3_resource_stub.client).to receive(:put_object)
+                                       .with(bucket: s3_bucket,
+                                             key: "#{s3_path_prefix}cabinet-office/2020-05-01.csv",
+                                             body: "cabinet-office-data")
+                                       .and_call_original
+
+        expect(s3_resource_stub.client).to receive(:put_object)
+                                       .with(bucket: s3_bucket,
+                                             key: "#{s3_path_prefix}third-party/2020-05-01.csv",
+                                             body: "third-party-data")
+                                       .and_call_original
+        described_class.call
+      end
+
+      it "notifies partners with the times and number of responses" do
+        expect(AskExport::PartnerNotifier)
+          .to receive(:call)
+          .with(since_time, until_time, responses.count)
+        described_class.call
+      end
+    end
+
+    context "before 10am" do
+      around do |example|
+        travel_to(Time.parse("2020-05-01 08:00 UTC")) { example.run }
```

Comment:
Could we do `Time.zone.parse("2020-04-30 09:59")` to save a bit of mental effort?

Reply:
Sure, I've made them all pretty consistent.